### PR TITLE
#2656: Fixed Settings page to show which team your in even when you have leadership role

### LINKED ISSUE #2656: User Settings team name
### Desired Changes

Currently if you are listed as a lead under a specific team, it doesn't show that on the team in the user settings page.

image

### Acceptance Criteria
- If a user is listed as a head or a lead under a team, it should reflect that in the user settings page

### Screenshots (as needed)

_No response_

### PR DESCRIPTION
**Pull request by elbow74**
## Changes

Fixed Settings page to show which team your in even when you have leadership role

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## Screenshots

Screenshot 2025-01-29 at 7 03 41 PM

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ x] All commits are tagged with the ticket number
- [ x] No linting errors / newline at end of file warnings
- [ x] All code follows repository-configured prettier formatting
- [ x] No merge conflicts
- [ x] All checks passing
- [ x] Screenshots of UI changes (see Screenshots section)
- [ x] Remove any non-applicable sections of this template
- [ x] Assign the PR to yourself
- [ x] No `yarn.lock` changes (unless dependencies have changed)
- [ x] Request reviewers & ping on Slack
- [ x] PR is linked to the ticket (fill in the closes line below)

Closes #2656

### CODE REVIEW

**Comment by vsp05** [comment]
LGTM! Great work! I'll ask sean to take a look at it once the issue with the tests is resolved

**Review by vsp05 — DISMISSED** [review]
great work! @walker-sean could you take a look when you get a chance?

**Review by jonathan-chen10 — DISMISSED** [review]
lgtm

File: src/frontend/src/pages/SettingsPage/SettingsDetails.tsx [hunk 1]
```diff
@@ -40,7 +40,12 @@ const SettingsDetails: React.FC = () => {
     return <LoadingIndicator />;
 
   const userTeams = teams.filter((team) =>
-    team.members.some((member) => member.userId === user.userId || team.head.userId === user.userId)
+    team.members.some(
```
Comment (walker-sean) [inline src/frontend/src/pages/SettingsPage/SettingsDetails.tsx]: The `some` calls its given callback for each item in the array, so we're potentially doing a lot of unnecessary checks here. Let's do it serially, checking if the user is in the team, then if the user is a lead, then if the user is the head.

**Review by walker-sean — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/SettingsPage/SettingsDetails.tsx (hunk 1 above)
Comment (Peyton-McKee) [reply src/frontend/src/pages/SettingsPage/SettingsDetails.tsx]: im pretty sure we have a userOnTeam function already that can be used

**Review by jonathan-chen10 — APPROVED** [review]
lgtm

**Review by walker-sean — APPROVED** [review]
LGTM
